include/Core/Drawable.hpp

```diff
@@ -21,22 +24,19 @@ constexpr int UNIFORM_SURFACE_LOCATION = 0;
 
 class Drawable {
 public:
-    Drawable(glm::mat3 transform);
 
-    void Draw();
+    virtual void Draw() = 0;
 
     int GetUniformSurfaceLocation() const { return UNIFORM_SURFACE_LOCATION; }
 
-private:
-    void InitProgram();
-    void InitVertexArray();
-    void InitUniformBuffer();
+protected:
+    glm::mat3 m_Transform;
 
-    static std::unique_ptr<Program> s_Program;
-    static std::unique_ptr<VertexArray> s_VertexArray;
-    static std::unique_ptr<UniformBuffer<Matrices>> s_UniformBuffer;
+    std::unique_ptr<SDL_Surface, std::function<void(SDL_Surface *)>> m_Surface =
+        nullptr;
+    std::unique_ptr<Texture> m_Texture = nullptr;
```

Comment:
Not every `Drawable` needs to have `SDL_Surface *` and `Texture`. These things are for when something needs to be an texture.

Reply:
I feel like if those static resources are defined in each classes, there isn't really a need for `Drawable` to have any member